plugins/modules/azure_rm_aks.py

```diff
@@ -520,6 +537,12 @@ def create_addon_profiles_spec():
 )
 
 
+api_server_access_profile_spec = dict(
+    authorized_ip_ranges=dict(type='list', elements='string'),
```

Comment:
There is no 'string' type in python!
```suggestion
    authorized_ip_ranges=dict(type='list', elements='str'),
```